src/bp/common/variables.ts

```diff
@@ -0,0 +1,80 @@
+import { BoxedVarContructor, BoxedVariable, NLU } from 'botpress/sdk'
+
+export class BaseVariable<T, V = any> implements BoxedVariable<T, V> {
+  protected _type: string
+  protected _enumType?: string
+  protected _value?: T
+  protected _confidence?: number
+  protected _nbTurns: number
+  protected _config?: V
+
+  protected _getEnumList: () => NLU.EntityDefOccurrence[]
+
+  constructor({ type, enumType, nbOfTurns, value, confidence, config, getEnumList }: BoxedVarContructor<T, V>) {
```

Comment:
enumType for BaseVariable ? does that mean that every variable will need to have an enumType ? if not then why is it listed in the BaseVariable ?

Reply:
Nop, but any variable "could" use it. To avoid using async calls in variables and keep them lightweight, we preload the types and make it accessible to any variable.

Reply:
My comment was more code styling, don't like the fact that the BaseVariable implementation has some things related to enums.

Enums are in fact just other variable types I would prefer to add properties to enum variables implementations and not the base variable

Reply:
There is a single variable implementation. Since enumType is handled by the core and the variable enum in a module, then it must be accessible to all variable types. Separating those would just complexify the code for no reason. 

When you create a variable, you can pass a property "enumType", and when you do, you can access the values using getEnumList.